src/humidity.jl

```diff
@@ -0,0 +1,63 @@
+"""
+Compute the saturation specific humidity for a single atmospheric level.
+"""
+function get_saturation_specific_humidity(
+    T::Array{NF,2},  # Absolute temperature for a single atmospheric level
+    p::Array{NF,2},  # Normalised surface pressure
+    σ::NF,           # σ of the current atmospheric level
+    M::ModelSetup{NF},
+) where {NF<:AbstractFloat}
+    @unpack nlon, nlat = M.geospectral.geometry
+    @unpack e₀, C₁, C₂, T₀, T₁, T₂ = M.params.humidity_coefs
+
+    Qsat = zeros(nlon, nlat)  # Saturation specific humidity
+
+    @inbounds for j = 1:nlat, i = 1:nlon
+        if T[i, j] > T₀
+            Qsat[i, j] = e₀ * exp(C₁ * (T[i, j] - T₀) / (T[i, j] - T₁))
```

Comment:
Do you think it would make sense to write this function already in ˚C rather than having `T` in Kelvin? Also to be more explicit we called temperature `temp` throughout other functions ...

Reply:
also what about `humid_saturation`? and then add the information in a comment that it's actually a water vapour pressure [hPa] or similar?

Reply:
> Do you think it would make sense to write this function already in ˚C rather than having `T` in Kelvin?

Just to be clear, do you mean specifically just write this line in Kelvin so that the function more closely resembles the standard Magnus form? i.e. precede this line with a line like `temp_celsius = temp - T0`?

Reply:
I mean rewriting the entire model such that temperature uses ˚C instead of Kelvin. I expect everything to work very similar, instead of probably radiation, where somewhere the Stefan-Boltzmann ~T^4 should appear. We could then convert from ˚C to Kelvin therein. But from a precision point-of-view using Kelvin is awful as you'll hit massive rounding errors in the time integration, e.g. T = 300K, tendency dT = 0.1K
```julia
julia> T = Float16(300)
Float16(300.0)

julia> dT = Float16(0.1)
Float16(0.1)

julia> T+dT
Float16(300.0)

julia> T2 = Float16(300-273.15)
Float16(26.84)

julia> T2+dT
Float16(26.94)
```
In Kelvin you can't resolve the increment, but in ˚C you can.

Reply:
Having said that, we obviously do the time stepping in spectral space, meaning that we'd only have that problem on the `l=m=0` mode which we could also solve with compensated summation, but I think in general, we should try to use ˚C if possible. If it turns out that ˚C is a bad idea then we can still revert back.

Reply:
Agreed it makes sense to use Celsius (or at the very least to try it out). I would definitely write that up as a separate issue though, and keep this PR in Kelvin for now.